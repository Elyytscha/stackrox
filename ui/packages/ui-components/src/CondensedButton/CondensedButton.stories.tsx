import React from 'react';
import { Meta, Story } from '@storybook/react/types-6-0';

import CondensedButton from './CondensedButton';

export default {
    title: 'CondensedButton',
    component: CondensedButton,
} as Meta;

export const DefaultButton: Story<{}> = () => {
    function onClick(): void {}
    return (
        <CondensedButton type="button" onClick={onClick}>
            Click me
        </CondensedButton>
    );
};
