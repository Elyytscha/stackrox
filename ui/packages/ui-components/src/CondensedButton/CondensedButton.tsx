import React, { ReactElement } from 'react';

import Button, { HOCButtonProps } from '../Button';

export type CondensedButtonProps = HOCButtonProps;

function CondensedButton({
    type,
    onClick,
    children,
    isDisabled,
}: CondensedButtonProps): ReactElement {
    return (
        <Button type={type} onClick={onClick} colorType="base" isDisabled={isDisabled} isCondensed>
            {children}
        </Button>
    );
}

export default CondensedButton;
