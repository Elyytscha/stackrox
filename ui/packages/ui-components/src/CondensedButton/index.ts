export { default } from './CondensedButton';
export * from './CondensedButton';
