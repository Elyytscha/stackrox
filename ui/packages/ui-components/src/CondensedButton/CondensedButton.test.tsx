import React from 'react';
import { render } from '@testing-library/react';

import CondensedButton from './CondensedButton';

describe('CondensedButton', () => {
    test('renders the button', () => {
        function onClick(): void {}

        const { container, getByText } = render(
            <CondensedButton type="button" onClick={onClick}>
                Click me!
            </CondensedButton>
        );

        expect(getByText('Click me!')).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });
});
