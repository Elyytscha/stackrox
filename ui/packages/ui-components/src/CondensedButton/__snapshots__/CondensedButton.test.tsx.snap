// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CondensedButton renders the button 1`] = `
<div>
  <button
    class="border-2 font-600 inline-flex items-center justify-center rounded-sm uppercase text-sm border-base-400 bg-base-100 hover:bg-base-200 hover:text-base-700 text-base-800 px-2"
    type="button"
  >
    Click me!
  </button>
</div>
`;
